src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -3370,6 +3370,25 @@ protected override ValueNode GetMethodParameterValue (MethodDefinition method, i
 				};
 			}
 
+			protected override ValueNode GetFieldValue (MethodDefinition method, ValueNode instanceValue, FieldDefinition field)
+			{
+				DynamicallyAccessedMemberKinds memberKinds = _flowAnnotations.GetFieldAnnotation (field);
+				return new LoadFieldValue (instanceValue, field, memberKinds) {
+					SourceContext = method
+				};
+			}
+
+			protected override void HandleStoreField (MethodDefinition method, FieldDefinition field, Instruction operation, ValueNode instanceValue, ValueNode valueToStore)
+			{
+				var requiredMemberKinds = _flowAnnotations.GetFieldAnnotation (field);
+				if (requiredMemberKinds != 0) {
+					// TODO: There's no way to represent store to a field given current ReflectionPatternContext (and the underlying IReflectionPatterRecorder)
```

Comment:
Maybe this can be handled the same as method return values? (Build a list of all assigned fields as we scan the body but process the list in `ScanAndProcessReturnValue`.)

Reply:
It could be, but that would require more allocations (the list of values and so on). I find the virtual method a bit cleaner personally.

Reply:
Sounds good. Maybe when addressing this TODO, we could unify this with return value handling (the `ret` instruction sounds like "a `stfld` into a pretend return value field").